File: src/main/kotlin/function/ShadowFunctionArgument.java
package function;

public class ShadowFunctionArgument {

    public static void main(String[] args) {
        String str = "Suresh";
        System.out.println(str);
        printValue(str);
        System.out.println(str);
    }

    private static void printValue(String str) {
        str = "Ramesh";
        System.out.println(str);
    }

}

File: src/main/kotlin/function/ShadowFunctionArgument.kt
package function

import Wrapper

fun main() {
    var str = "123"
    printValue(str)
    println(str)

    val wrapper = Wrapper("Suresh")
    println(wrapper.value)
    printValue(wrapper)
    println(wrapper.value)
}

private fun printValue(str: String) {
    var str = str
    str = "abc"
    println(str)
}

private fun printValue(str: Wrapper<String>) {
    str.value = "Ramesh"
    println(str.value)
}

